pkg/log/printer.go

```diff
@@ -0,0 +1,11 @@
+package log
+
+import (
+	"context"
+)
+
+type Printer interface {
```

Comment:
At this point, it makes sense to rename this interface as `Logger`, since now it has more than just the `Print` function
```suggestion
type Logger interface {
```